service/src/main/java/bio/terra/pipelines/stairway/imputation/steps/gcp/SubmitCromwellSubmissionStep.java

```diff
@@ -59,10 +71,71 @@ public StepResult doStep(FlightContext flightContext) {
     String description = inputParameters.get(JobMapKeys.DESCRIPTION.getKeyName(), String.class);
     String wdlMethodName =
         inputParameters.get(RunImputationJobFlightMapKeys.WDL_METHOD_NAME, String.class);
+    String wdlMethodVersion =
+        inputParameters.get(RunImputationJobFlightMapKeys.WDL_METHOD_VERSION, String.class);
+
+    List<PipelineInputDefinition> inputDefinitions =
+        inputParameters.get(
+            RunImputationJobFlightMapKeys.PIPELINE_INPUT_DEFINITIONS, new TypeReference<>() {});
+    List<PipelineOutputDefinition> outputDefinitions =
+        inputParameters.get(
+            RunImputationJobFlightMapKeys.PIPELINE_OUTPUT_DEFINITIONS, new TypeReference<>() {});
 
     // validate and extract parameters from working map
     FlightMap workingMap = flightContext.getWorkingMap();
 
+    MethodConfiguration methodConfiguration;
+    try {
+      // grab current method config and validate it
+      methodConfiguration =
+          rawlsService.getCurrentMethodConfigForMethod(
+              samService.getTeaspoonsServiceAccountToken(),
+              controlWorkspaceProject,
+              controlWorkspaceName,
+              wdlMethodName);
+    } catch (RawlsServiceApiException e) {
+      // if we fail to grab the method config then retry
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
+    }
+    boolean validMethodConfig =
+        rawlsService.validateMethodConfig(
+            methodConfiguration,
+            wdlMethodName,
+            inputDefinitions,
+            outputDefinitions,
+            wdlMethodVersion);
+
+    // if not a valid method config, set the method config to what we think it should be.  This
+    // shouldn't happen
+    if (!validMethodConfig) {
+      logger.warn(
+          "found method config that was not valid for billing project: {}, workspace: {}, method name: {}, methodConfigVersion: {}",
```

Comment:
nit: it's not necessarily "not valid" but it's not matching our expectations. super minor but to me "not valid" suggests there's an actual error in the setup

Reply:
but to us there is an error in the setup no?